crypto/cmp/cmp_util.c

```diff
@@ -100,8 +100,8 @@ const char *ossl_cmp_log_parse_metadata(const char *buf,
                     *file = OPENSSL_strndup(p_file, p_line - 1 - p_file);
                     /* no real problem if OPENSSL_strndup() returns NULL */
                     *line = (int)line_number;
-                    msg = strchr(p_level, ':') + 1;
-                    if (*msg == ' ')
+                    msg = strchr(p_level, ':');
+                    if (msg != NULL && *(msg++) == ' ')
```

Comment:
No, `*++msg`.

Reply:
urgh I thought the ( ) would give me precedence over the dereference, irrespective or pre/post increment

Reply:
Regardless in which expression context the post-{inc,de}rement operator is used, its value is the variable contents _before_.